miden-crypto/src/merkle/smt/full/large/storage/rocksdb.rs

```diff
@@ -0,0 +1,1017 @@
+    /// storage.
+    ///
+    /// This method sets up the necessary column families (`leaves`, `subtrees`, `metadata`)
+    /// and applies various RocksDB options for performance, such as caching, bloom filters,
+    /// and compaction strategies tailored for SMT workloads.
+    ///
+    /// # Errors
+    /// Returns `StorageError::BackendError` if the database cannot be opened or configured,
+    /// for example, due to path issues, permissions, or RocksDB internal errors.
+    pub fn open(path: &PathBuf) -> Result<Self, StorageError> {
+        // Base DB options
+        let mut db_opts = Options::default();
+        db_opts.create_if_missing(true);
+        db_opts.create_missing_column_families(true);
+        db_opts.increase_parallelism(rayon::current_num_threads() as i32);
+        db_opts.set_max_open_files(1024);
+        db_opts.set_max_background_jobs(rayon::current_num_threads() as i32);
+
+        // Share a 1 GiB LRU block cache
+        let cache = Cache::new_lru_cache(1 << 30);
+
+        // Common table options for bloom filtering and cache
+        let mut table_opts = BlockBasedOptions::default();
+        table_opts.set_block_cache(&cache);
+        table_opts.set_bloom_filter(10.0, false);
+        table_opts.set_whole_key_filtering(true);
+        table_opts.set_pin_l0_filter_and_index_blocks_in_cache(true);
+
+        // Column family for leaves
+        let mut leaves_opts = Options::default();
+        leaves_opts.set_block_based_table_factory(&table_opts);
+        leaves_opts.set_write_buffer_size(256 << 20);
+        leaves_opts.set_max_write_buffer_number(6);
+        leaves_opts.set_min_write_buffer_number_to_merge(2);
+        leaves_opts.set_compaction_style(DBCompactionStyle::Level);
+        leaves_opts.set_target_file_size_base(64 << 20);
+        leaves_opts.set_compression_type(DBCompressionType::Lz4);
+
+        // Helper to build subtree CF options with correct prefix length
+        fn subtree_cf(cache: &Cache, prefix_bytes: usize) -> Options {
+            let mut tbl = BlockBasedOptions::default();
+            tbl.set_block_cache(cache);
+            tbl.set_bloom_filter(16.0, true);
+            tbl.set_whole_key_filtering(true);
+            tbl.set_pin_l0_filter_and_index_blocks_in_cache(true);
+
+            let mut opts = Options::default();
+            opts.set_prefix_extractor(SliceTransform::create_fixed_prefix(prefix_bytes));
+            opts.set_block_based_table_factory(&tbl);
+            opts.set_write_buffer_size(64 << 20);
+            opts.set_max_write_buffer_number(6);
+            opts.set_min_write_buffer_number_to_merge(2);
+            opts.set_compaction_style(DBCompactionStyle::Level);
+            opts.set_level_zero_file_num_compaction_trigger(4);
+            opts.set_target_file_size_base(256 << 20);
+            opts.set_compression_type(DBCompressionType::Lz4);
+            opts
+        }
```

Comment:
I still don't know what these magic numbers mean or which are important.  Why does the Subtree column have a higher bloom filter than the common table options? `256 << 20` and `64 << 20` feel oddly specific, where did those come from? 6 write buffers also feels pretty specific.

Rocksdb doesn't even have `set_prefix_extractor()` or `set_block_based_table_factory()` documented (what the hell), so those ones *definitely* needs a comment.